src/system.ts

```diff
@@ -411,6 +415,12 @@ export const chunkDataSource =
         chunkDataStore: new FsChunkDataStore({ log, baseDir: 'data/chunks' }),
       });
 
+export const chunkDataSource =
+  config.CHUNK_DATA_SOURCE_TYPE === 'legacy-s3' &&
+  config.CHUNK_METADATA_SOURCE_TYPE === 'legacy-psql'
+    ? new CompositeChunkSource(chunkMetaDataSource, basicChunkDataSource)
+    : basicChunkDataSource;
```

Comment:
`basicChunkDataSource` won't have the right return type once other comments are taken into account, so we shouldn't fall back to it here.